scripts/generate_schema/worldbank/generate_worldbank_schema.py

```diff
@@ -0,0 +1,126 @@
+import os
+from argparse import ArgumentParser
+import requests
+import json
+
+DEFAULT_KEY = {
+
+}
+
+
+def getAllIndicatorList():
+    url = "https://api.worldbank.org/v2/indicators?format=json&page=1&per_page=2"
+    res = requests.get(url)
+    data = res.json()
+    return data[1]
+
+
+def generate_json_schema(dst_path):
+    unique_urls_str = getAllIndicatorList()
+    # print(unique_urls_str)
+    for commondata in unique_urls_str:
+        # print('here',commondata)
+        # commondata=data[0]
+        # print(commondata['id'])
+        urldata = "https://api.worldbank.org/v2/countries/br/indicators/" + commondata['id'] + "?format=json"
+        resdata = requests.get(urldata)
+        data_ind = resdata.json()
+        # print(data_ind)
+        materialiseFormat = 'csv'
+        infoFormat = 'json'
+        print("Generating schema for Trading economics", commondata['name'])
+        schema = {}
+        schema["title"] = commondata['name']
+        schema["description"] = commondata['sourceNote']
+        schema["url"] = "https://api.worldbank.org/v2/indicators/" + commondata['id'] + "?format=json"
+        schema["keywords"] = [i for i in commondata['name'].split()]
+        schema["date_updated"] = data_ind[0]["lastupdated"]
+        schema["license"] = None
+        schema["provenance"] = "http://worldbank.org"
+        schema["original_identifier"] = commondata['id']
+
+        schema["materialization"] = {
+            "python_path": "worldbank_materializer",
+            "arguments": {
+                "url": "https://api.worldbank.org/v2/indicators/" + commondata['id'] + "?format=json"
```

Comment:
You are recording the url of this dataset for the particular indicator, which is good. 
But in the materializer: https://github.com/usc-isi-i2/datamart/blob/64248c5483ba85fba82730beb71a631130f879a1/datamart/materializers/worldbank_materializer.py#L31
You are trying to get dataset id from constrains. The metadata dictionary is the metadata for this dataset. So you should get the url from its `materialization. arguments.url` as you recorded here. Then add date and location constrains to this url. Instead of adding here: https://github.com/usc-isi-i2/datamart/blob/64248c5483ba85fba82730beb71a631130f879a1/datamart/materializers/worldbank_materializer.py#L47

Reply:
Okay so you suggest, I collect dataset url from the schema and not constrains. Will do that